testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/ssl/UndertowSSLv2HelloTestCase.java

```diff
@@ -0,0 +1,373 @@
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.AssumptionViolatedException;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runner.RunWith;
+import org.junit.runners.model.Statement;
+import org.wildfly.security.auth.client.AuthenticationContext;
+import org.wildfly.security.auth.client.ElytronXmlParser;
+import org.wildfly.security.auth.client.InvalidAuthenticationConfigurationException;
+import org.wildfly.test.integration.elytron.util.WelcomeContent;
+import org.wildfly.test.security.common.AbstractElytronSetupTask;
+import org.wildfly.test.security.common.elytron.ConfigurableElement;
+import org.wildfly.test.security.common.elytron.CredentialReference;
+import org.wildfly.test.security.common.elytron.Path;
+import org.wildfly.test.security.common.elytron.SimpleKeyManager;
+import org.wildfly.test.security.common.elytron.SimpleKeyStore;
+import org.wildfly.test.security.common.elytron.SimpleServerSslContext;
+import org.wildfly.test.security.common.elytron.SimpleTrustManager;
+
+/**
+ * Smoke test for two way SSL connection with Undertow HTTPS listener backed by Elytron
+ * server-ssl-context
+ * with need-client-auth=true (client certificate is required) using SSLv2Hello.
+ *
+ * In case the client certificate is not trusted or present, the SSL handshake should fail.
+ *
+ * Additionally, smoke test for one way SSL connection with Undertow HTTPS listener
+ * backed by Elytron server-ssl-context.
+ *
+ * @author Ondrej Kotek
+ * @author Sonia Zaldana
+ */
+@RunWith(Arquillian.class)
+@ServerSetup({ UndertowSSLv2HelloTestCase.ElytronSslContextInUndertowSetupTask.class, WelcomeContent.SetupTask.class})
+@RunAsClient
+public class UndertowSSLv2HelloTestCase {
+
+    private static final String NAME = UndertowTwoWaySslNeedClientAuthTestCase.class.getSimpleName();
+    private static final File WORK_DIR = new File("target" + File.separatorChar +  NAME);
+    private static final File SERVER_KEYSTORE_FILE = new File(WORK_DIR, SecurityTestConstants.SERVER_KEYSTORE);
+    private static final File SERVER_TRUSTSTORE_FILE = new File(WORK_DIR, SecurityTestConstants.SERVER_TRUSTSTORE);
+    private static final String PASSWORD = SecurityTestConstants.KEYSTORE_PASSWORD;
+    private static final String SSLV2HELLO_CONTEXT = "SSLv2HelloContext";
+    private static final String DEFAULT_CONTEXT = "DefaultContext";
+    private static final String SSLV2HELLO_CONTEXT_ONE_WAY = "SSLv2HelloContextOneWay";
+    private static final String HTTPS = "https";
+    private static URL securedRootUrl;
+    public static String disabledAlgorithms;
+
+    @ClassRule
+    public static IbmVerification ibmVerification =
+            new IbmVerification(System.getProperty("java.vendor").contains("IBM"));
```

Comment:
This is not critical at all, but there is a `TestSuiteEnvironment.isIbmJvm()` or `TestSuiteEvronment.isJ9Jvm()` is case this is supported on OpenJ9.

Reply:
Added ``TestSuiteEnvironment.isIbmJvm()`` instead.